### ISSUE TITLE
"Checksum does not match" when kwargs parameter is not defined on class

### ISSUE THREAD

**Issue by frbor** [adamghill/django-unicorn#140]
I upgraded to 0.18.0, but started to see responses with

```javascript
{"error": "Checksum does not match"}
```
on one of the pages.

My problem was a component like this:

```python
class MyView(UnicornView):
    text = "hello"
    active = None

    def __init__(self, *args, **kwargs):
        super().__init__(**kwargs)
        self.active = kwargs.get("active_tab", "default")

```

In the data sent I saw both `active` and `active_tab`.

It was an easy fix for me (I just made sure the parameter in kwargs was named the same as the class variable), but I thought other might get into the same problem, so i'm reporting it as a bug.

You can see the problem in action in [this branch](https://github.com/frbor/django-unicorn/tree/kwargs/example) in the validation example:

- [https://github.com/frbor/django-unicorn/blob/kwargs/example/unicorn/components/validation.py](https://github.com/frbor/django-unicorn/blob/kwargs/example/unicorn/components/validation.py)
- [https://github.com/frbor/django-unicorn/blob/kwargs/example/www/templates/www/validation.html](https://github.com/frbor/django-unicorn/blob/kwargs/example/www/templates/www/validation.html)

I'm sorry if I overlooked any of this in the documentation, but I'm not sure whether this should work? 

There might also be other reasons to send a parameter to kwargs without assigning it to the component?

On a side note I have had a couple of issues previously with the "Checksum does not match" error. I'm not sure *what* is checksumed, so might add that to the documentation some place?

Anyhow, impressive progress on the project. It has come a long way so far and keeps getting better!

**Comment by adamghill** [comment]
Thanks for the bug report (again!). I can replicate this issue, so hopefully I can fix this pretty quickly.

I will try to document the "Checksum does not match" error somewhere as well. I added some code to potentially alleviate it, but it basically happens when a page "gets out of sync" and `Unicorn` can't figure out what the correct state should be. I'll continue to try to reduce how frequently that happens as I see it in testing.

**Comment by adamghill** [comment]
I just released 0.18.1 which should allow component kwargs to differ from component attributes again. Closing for now, but I added a note to https://github.com/adamghill/django-unicorn.com/issues/2 to document checksums somewhere in the docs.

**Comment by Bogdan-Branzaniuc** [comment]
I am running into some issues with this in 0.51.1.

      class ProductView(UnicornView):
          bag = None
          bag_keys = None
          products = Product.objects.all()
          def __init__(self, *args, **kwargs):
              super().__init__(**kwargs)  # calling super is required
      
          def ad_or_remove_from_bag(self):
              self.bag = self.request.session.get('bag', {})
              self.bag_keys = list(int(key) for key in self.bag.keys())
              print(self.bag)
              
now, I am trying to call this view using a simple button  with an attribute of u:click="ad_or_remove_from_bag"

And getting the error Checksum does not match. 

However it does work once whenever I refresh the session in django, but then it completely stops and throws the error. in previous projects the error appears too, but only when I spam the buttons. 

any Idea how and why this happens ? 

Any help is much appreciated
